docs/README.md

```diff
@@ -900,15 +900,132 @@ where the schema for `CreateProductInput` is
 
 ### 2. Events
 
+Events are **immutable facts** within your application's domain. They are the cornerstone of Booster because of its event-driven and event-sourced nature. Booster events are TypeScript classes decorated with `@Event`. An event class may looks like this:
+
+```typescript
+@Event
+export class EventName {
+  public constructor(readonly field1: SomeType,
+                     readonly field2: SomeOtherType,
+                     /* This event's entity ID must be present */) {}
+
+  public entityID(): UUID {
+    return /* the associated entity ID */
+  }
+}
+```
+
+Events and [entities](#4-entities-and-reducers) are closely related because entities represent the application's state after reducing the stream of events. All the events associated with the same entity are reduced together in the order they have occurred. If an event is related to an entity or not, it is domain-dependent. That's why Booster requires you to implement the `entityID` method. Entities and events are somehow coupled, but that coupling is limited to data. Take a look at this event:
+
+```typescript
+@Event
+export class CartPaid {
+  public constructor(
+    readonly cartID: UUID,
+    readonly paymentID: UUID) {}
+
+  public entityID(): UUID {
+    // returning cartID or paymentID depends on the business logic
+    return this.paymentID
+  }
+}
+```
+
+Note:
+
+> An event must be able to find the entity ID value within the arguments. Or well because it got injected directly, or because it is present as a nested attribute. In the case of the `CartPaid` example, the entity ID is injected directly.
```

Comment:
```suggestion
> An event must be able to find the entity ID value within the event attributes, either among the parameters injected in the constructor or because it is present in a nested attribute of some of the event attributes. In the case of the `CartPaid` example, the entity ID (`paymentID`) is injected directly.
```